express-zod-api/src/documentation-helpers.ts

```diff
@@ -723,27 +724,26 @@ export const onMissing: SchemaHandler<
 };
 
 export const excludeParamsFromDepiction = (
-  depicted: SchemaObject | ReferenceObject,
+  subject: SchemaObject | ReferenceObject,
   names: string[],
-): SchemaObject | ReferenceObject => {
-  if (isReferenceObject(depicted)) return depicted;
-  const copy = { ...depicted };
-  if (copy.properties) copy.properties = R.omit(names, copy.properties);
-  if (copy.examples)
-    copy.examples = copy.examples.map((entry) => R.omit(names, entry));
-  if (copy.required)
-    copy.required = copy.required.filter((name) => !names.includes(name));
-  if (copy.allOf) {
-    copy.allOf = copy.allOf.map((entry) =>
-      excludeParamsFromDepiction(entry, names),
-    );
-  }
-  if (copy.oneOf) {
-    copy.oneOf = copy.oneOf.map((entry) =>
-      excludeParamsFromDepiction(entry, names),
-    );
-  }
-  return copy;
+): [SchemaObject | ReferenceObject, boolean] => {
+  if (isReferenceObject(subject)) return [subject, false];
+  let hasRequired = false;
+  const subTransformer = R.map((entry: SchemaObject | ReferenceObject) => {
+    const [sub, subRequired] = excludeParamsFromDepiction(entry, names);
+    hasRequired = hasRequired || subRequired;
+    return sub;
+  });
```

Comment:
this probably should not really go deeper than one level

```ts
type TheCase = {
  one?: { sub: true }
} | {
  two?: { sub: true }
};
```

should be no required in this case